TestAll.ps1

```diff
@@ -0,0 +1,76 @@
+<#  
+.SYNOPSIS
+    Run the TAEF dlls generated by WinAppSDK
+
+.DESCRIPTION
+    The TestAll script will take the folder input and look for subfolders containing a .testmd file. WinAppSDK
+    components define a testmd with the following schema and runs the TAEF dll in the subfolder.
+    
+    Example:
+    {
+        "Tests": [
+            {
+                "Description": "Enabled: This module tests the push notifications component in WinAppSDK.",
+                "Filename": "PushNotificationTests.dll",
```

Comment:
In the PR description you said filename and architecture are mandatory, consider mentioning that here too.

Reply:
I don't think I'm allowed to add a comment to JSON. Maybe I can add a sample testdef file to the repo for components to copy/paste?

Reply:
> I don't think I'm allowed to add a comment to JSON. Maybe I can add a sample testdef file to the repo for components to copy/paste?

Correct, [JSON](https://www.json.org/json-en.html) has no 'comment' concept. That's what Description would do

Also nothing stops you from adding other data if the parser is forgiving (not strict) e.g.

```
{
    "Tests": [
        {
            "Description": "Enabled: This module tests the push notifications component in WinAppSDK.",
            "Filename": "PushNotificationTests.dll",
            "Parameters": "",
            "Architectures": ["x86", "x64", "arm64"],
            "Status": "Enabled",
            "Warning": "Danger Will Robinson! Danger!",
            "Room Service": "Waffles, Omelette and Pancakes available. Call 1-800-HUNG-RY1"
        }
    ]
}
```

>Maybe I can add a sample testdef file to the repo for components to copy/paste?

No need because you're going to add a _working real-world_ `Test.testdef`(?) to `tools\ProjectTemplates\test.cpp.dll.taef\` right? This PR or follow on, same diff

```
{
    "Tests": [
        {
            "Description": "PurojekutoTenpuret tests for feature Tokucho",
            "Filename": "PurojekutoTenpuret.dll",
            "Parameters": "",
            "Architectures": ["x86", "x64", "arm64"],
            "Status": "Enabled",
        }
    ]
}
```

Reply:
Done.